.github/workflows/bump_torch_mlir.yml

```diff
@@ -0,0 +1,67 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# Automatically bumps the torch-mlir submodule to match shark-infra/torch-mlir:main.
+# Manually triggered for now, but can be scheduled in the future.
+# This typically takes ~90s to run.
+name: Bump submodule - shark-infra/torch-mlir
+
+on:
+  # TODO: Uncomment the following lines to enable the schedule when this action is stable & useful enough
+  # schedule:
+  #   - cron: '0 0 * * 0,4'  # Run at 00:00 on Sunday and Thursday
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths:
+      - .github/workflows/bump_torch_mlir.yml
+
+jobs:
+  bump_submodule:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c  # v3.3.0
+        with:
+          submodules: true
```

Comment:
(fine to ignore)

We could limit this to just updating the `third_party/torch-mlir` submodule
https://github.com/renxida/iree/actions/runs/8912266204/job/24475350099 took 40 seconds to checkout submodules (20 seconds for LLVM)

https://stackoverflow.com/questions/16728866/how-to-only-update-specific-git-submodules

That matters less here for a workflow that is already fast and run infrequently though, so the simplicity in the workflow file may be worth some wasted time (it does matter more for Windows build/test presubmit CI and such, where checkout out the repo can take 10 minutes x_x)